contracts/TokenConverter/RiskFundConverter.sol

```diff
@@ -208,16 +208,16 @@ contract RiskFundConverter is AbstractTokenConverter {
         uint256 balance = IERC20Upgradeable(tokenAddress).balanceOf(address(this));
         uint256 balanceDiff = balance - assetsReserves[tokenAddress];
 
-        uint256 amountDiff = amount - balanceDiff;
+        int256 amountDiff = int256(amount) - int256(balanceDiff);
```

Comment:
I think we could avoid the castings this way:

```
        if (balanceDiff > amount) {
            uint256 amountDiff = amount - balanceDiff;
            address[] memory pools = getPools(tokenAddress);
            uint256 assetReserve = assetsReserves[tokenAddress];
            for (uint256 i; i < pools.length; ++i) {
                uint256 poolShare = (poolsAssetsReserves[pools[i]][tokenAddress] * EXP_SCALE) / assetReserve;
                if (poolShare == 0) continue;
                updatePoolAssetsReserve(pools[i], tokenAddress, amountDiff, poolShare);
            }
            assetsReserves[tokenAddress] -= amountDiff;
        }
```

Reply:
Refactored, https://github.com/VenusProtocol/protocol-reserve/pull/17/commits/fd201247db8c1e4ee6b6dcbdf3ee37ff05f0b23e